src/cryptokit.ml

```diff
@@ -2072,6 +2074,98 @@ let new_key ?(rng = Random.secure_rng) ?e numbits =
 
 end
 
+module Paillier = struct
+
+  type key =
+    { size: int;
+      n: string;
+      n2: string;
+      g: string;
+      p: string;
+      q: string;
+      lambda: string;
+      mu: string}
+  let wipe_key k =
+    wipe_string k.n;
+    wipe_string k.n2;
+    wipe_string k.g;
+    wipe_string k.p;
+    wipe_string k.q;
+    wipe_string k.lambda;
+    wipe_string k.mu
+
+  let encrypt key msg ?r ()=
```

Comment:
I don't like this `r` optional parameter.  It makes it difficult for the user to choose a RNG other than `Random.secure_rng`, but too easy for them to shoot themselves in the foot by passing a fixed `r`.

In my proposed change, this becomes
```
  let encrypt ?(rng = Random.secure_rng) key msg =
```
and the remainder of the  function can be simplified a bit, since `r` is always randomly generated locally.
```

Reply:
@xavierleroy So basically i made this impl keeping in mind of my current project. There it was needed for a different purpose. I will change the structure to the proposed one so the arguments itself is changed.